src/Runner/Parallel/ProcessFactory.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Runner\Parallel;
+
+use PhpCsFixer\Runner\RunnerConfig;
+use React\EventLoop\LoopInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Process\PhpExecutableFinder;
+
+/**
+ * @author Greg Korba <greg@codito.dev>
+ *
+ * @internal
+ */
+final class ProcessFactory
+{
+    private InputInterface $input;
+
+    public function __construct(InputInterface $input)
+    {
+        $this->input = $input;
+    }
+
+    public function create(
+        LoopInterface $loop,
+        RunnerConfig $runnerConfig,
+        ProcessIdentifier $identifier,
+        int $serverPort
+    ): Process {
+        $commandArgs = $this->getCommandArgs($serverPort, $identifier, $runnerConfig);
+
+        return new Process(
+            implode(' ', $commandArgs),
+            $loop,
+            $runnerConfig->getParallelConfig()->getProcessTimeout()
+        );
+    }
+
+    /**
+     * @return list<string>
+     */
+    public function getCommandArgs(int $serverPort, ProcessIdentifier $identifier, RunnerConfig $runnerConfig): array
```

Comment:
private?

Reply:
I would need to modify `WorkerCommandTest` 😝.  Yeah, basically it's not used anywhere in `src` outside of the `ProcessFactory`, so I'll look at it. Anyway, as this is a factory, I believe this can be part of factory's API, so both acceptable for me.

Reply:
at least put `@internal` annotation pls

Reply:
Whole `ProcessFactory` is marked as `@internal`, so probably not needed to do more than that?